player/loadfile.c

```diff
@@ -835,6 +835,8 @@ void mp_deselect_track(struct MPContext *mpctx, struct track *track)
 {
     if (track && track->selected) {
         for (int t = 0; t < num_ptracks[track->type]; t++) {
+            if (track->type == STREAM_SUB && mpctx->current_track[t][track->type] != track)
```

Comment:
why hardcode the type here?

Reply:
Well it's only relevant for subs in the first place since that's the only type that can have more than one track. It could probably be removed though.

Reply:
it should be, because with `num_ptracks[]` the code is already generic

Reply:
> it should be, because with `num_ptracks[]` the code is already generic

type check is removed.  I add this type check before submit PR for safety , just in case if there is something wrong ^_^(not familiar with mpv codebase, didn't dig deep into the code, just use debug tool to find the bug and fix it )

Reply:
Could you squash the two commits together?